guidelines/design-guidelines.md

```diff
@@ -0,0 +1,36 @@
+## Configuration API's
+
+All configuration API's should be code first since that allows us to best guide the users and allows us to evolve the API's using obsolete messages.
+
+Configuration API's should be variable based and not returning `this` or the same instance it extends and therefore not chainable.
+
+Example:
+
+```
+var myConfig = endpointConfig.SomethingNeedingConfig()
+
+myConfig.SomeOption(X);
+myConfig.SomeOtherOption(X);
+
+```
+
+We prefer this over lambda based API's since:
```

Comment:
"lambda" should really be "delegate". A lambda is just one syntax for expressing a delegate. Perhaps:

> We prefer this over delegate-based (e.g. lambda expression) API's because:

There are also several instances of "lambda" further down in the document which should really be "delegate".